### ISSUE TITLE
Explore package testing possibilities & push auto updates on commit

### ISSUE THREAD

**Issue by theChaosCoder** [theChaosCoder/avsrepo#8]
One idea is to check if a commited update / new package is valid and then auto push a new compiled avspackages.json. Makes more sense if more people will commit something.

That means it has to be checked whether
- avspackages.json can be compiled without errors
- the latest commit can be "installed" ( hash + url must be valid)

**Comment by camjac251** [comment]
Would it be ideal to host it all over GitHub possibly? It could be done through Actions sending updates to the same release every time a package is changed. Then the URL would always be constant and could be served directly from GitHub

**Comment by theChaosCoder** [comment]
Yes, after the validation & push part is implemented, avspackages.json + ideally a simple web page could be hosted on github.

**Comment by camjac251** [comment]
Ah right. GitHub pages would be easier.

**Comment by camjac251** [comment]
I'm thinking of getting a test up and running today of `avsupdaterepo.py` running on a cron every 15 minutes through Actions. I think a better solution would be to ask the various people who are hosting their files on a GitHub repository, for access to this specific API call https://help.github.com/en/actions/reference/events-that-trigger-workflows#external-events-repository_dispatch

With this, it would send the needed trigger (alternative is a cron) to run the package update.

**Comment by theChaosCoder** [comment]
I don't think this can work reliably. vsrepo had always some problems with "bigger" plugins where the folder structure changed or other things got renamed.

**Comment by camjac251** [comment]
I was thinking in the end, each package could have its own action. So the triggers could occur for releases or commits to the Plugin's repository. Could I reach you another way by chance? Do you have telegram? Discord?

**Comment by theChaosCoder** [comment]
Discord: ChaosKing#4940
